operator/controller/src/main/java/io/apicurio/registry/operator/resource/app/AppDeploymentResource.java

```diff
@@ -67,11 +68,22 @@ protected Deployment desired(ApicurioRegistry3 primary, Context<ApicurioRegistry
                 .map(AppSpec::getFeatures)
                 .map(AppFeaturesSpec::getAllowDeletes)
                 .orElse(Boolean.FALSE);
+
         if (allowDeletes) {
             addEnvVar(envVars, new EnvVarBuilder().withName(EnvironmentVariables.APICURIO_REST_DELETION_ARTIFACT_VERSION_ENABLED).withValue("true").build());
             addEnvVar(envVars, new EnvVarBuilder().withName(EnvironmentVariables.APICURIO_REST_DELETION_ARTIFACT_ENABLED).withValue("true").build());
             addEnvVar(envVars, new EnvVarBuilder().withName(EnvironmentVariables.APICURIO_REST_DELETION_GROUP_ENABLED).withValue("true").build());
         }
+        boolean authEnabled = Optional.ofNullable(primary.getSpec().getApp())
+                .map(AppSpec::getAppAuthSpec)
+                .map(AppAuthSpec::getAuthEnabled)
+                .orElse(Boolean.FALSE);
+
+        //Configure auth when it's enabled
+        if (authEnabled) {
+            configureAuth(envVars, requireNonNull(ofNullable(primary.getSpec().getApp())
```

Comment:
Wouldn't this throw a NPE in the default case? I would add a few more validation checks for all fields in AppAuthSpec.

Reply:
No, authEnabled would be false.